### ISSUE TITLE
i18n: add support for no paragraph indents, standardized bottom margins

### ISSUE THREAD

**Issue by ptmkenny** [FriendsOfEpub/Blitz#44]
I'm interested in adding better support for the Japanese language (and any other languages with similar characteristics).

Japanese can be written horizontally or vertically.  I'm interested in adding horizontal support because that's closer to English, which Blitz already supports.

But, one major difference between Japanese and English is that paragraphs are not indented. Instead, a blank line is used between paragraphs.

So, I'd like to find a way to:

1. Eliminate paragraph indents and use an empty line between paragraphs.
2. Ensure a consistent top/bottom margin between elements.

For example:

* Paragraph
* One blank line
* Paragraph
* One blank line
* List item
* One blank line
* List item
* One blank line
* Table
* One blank line
* Paragraph
* One blank line
* Heading
* One blank line
* Paragraph

And so on.

I don't know too much about LESS but would it be possible to add something like a variable to toggle "paragraphs with indents and standard Blitz margins" or "no indents for paragraphs; use a blank line instead and ensure that a blank line is consistently used as a bottom margin between elements"?

**Comment by andrearicci** [comment]
you don't need a vaiable: just apply the modified style to paragraphs that have 'lang="ja"' - wich can be hinerithed from html itself.
in the css the selctor can be 
[yourtag]:lang(ja) { css declarations;} 
Setting the correct language shall also help the reader to choose the right dictionary.

**Comment by ptmkenny** [comment]
Thanks, that's a good way to style paragraph elements for a specific language.

But what I'm asking about is adding a way to standardize the bottom margins on all elements, including headings, tables, lists, and paragraphs.

**Comment by andrearicci** [comment]
never tested, but for vertical it shoul be:
```
*:lang(ja) {
    writing-mode: vertical-rl;
 }
```

**Comment by JayPanoz** [comment]
@ptmkenny 

> But what I'm asking about is adding a way to standardize the bottom margins on all elements, including headings, tables, lists, and paragraphs.

There is a `rhythm()` mixin available, whose [doc entry](https://friendsofepub.github.io/Blitz/documentation/docs.html#less-doc) is:

> .rhythm(@ font-scale, @ margin-top, @ margin-bottom)
> Computes a font-size, a line-height and vertical margins for the element. Unlike .fs (@ font-scale), it enforces vertical rhythm.
> The @ font-scale argument must be an integer (zero, positive or negative) telling which step in the typographic scale the element should use.
> The @ margin-top and @ margin-bottom arguments can be an integer or a decimal (zero or positive) telling the number of lines which should be used for those margins.

It’s used in typo.less for instance:

https://github.com/FriendsOfEpub/Blitz/blob/5dade12ba8574e96a1ec9cd99e54185a7230284b/Blitz_framework/LESS/base/typo.less#L53-L60

It is meant to recompute the proper `em` value for margins depending on the `font-size` of the element, and 1 line = 1 `line-height` – because some apps don’t support `rem` properly so we can’t use it.

It’s probably usable for this normalisation but may end up being a little bit verbose as you’ll have to handle all elements you want to cover in CSS selectors. That said, I don’t necessarily see another option, given `ems` depends on the `font-size`.

**Comment by andrearicci** [comment]
@ptmkenny Hi, actually I'm just a CSS hobbist with a Japanese wife. I've not that experience. My wife says that all the Japanese novels she reads on the kindle are vertically formatted. But I can't tell you how.
I can give you a link to free Japanese novel, where you can download some samples and check their CSS. (This is the publisher that you can download for free on amazon.jp). 
[https://www.aozora.gr.jp/](https://www.aozora.gr.jp/)
[by category](http://yozora.main.jp/)
[alphabetical order](https://www.aozora.gr.jp/index_pages/index_top.html#work_list)

Good luck

**Comment by ptmkenny** [comment]
@andrearicci Ok, thanks.  I can do vertical formatting; that is not the problem.

The issue is that the Amazon ecosystem apparently does not support lang-specific CSS.

**Comment by JayPanoz** [comment]
@ptmkenny 

Out of curiosity, do your elements have both `xml:lang` and `lang` attributes? E.g.

```
<html xml:lang="ja" lang="ja">
```

Reminds me of the issue with nav list you committed a few days ago, with Kindle ignoring the CSS namespace. 

And there are like 2 options here:

1. Kindle is handling XHTML as HTML (and CSS epub namespace + `xml:lang` fail in this context)
2. They are simply ignored during conversion.

**Comment by ptmkenny** [comment]
After some debugging, I found the root cause of this issue: pandoc, which I was using to generate the ebook file, was not applying the lang tag correctly even though I had specified it correctly in a YAML block.  This has been fixed from pandoc 2.8.1.

The reason why it worked in iBooks but not in Kindle is that iBooks reads the language attribute (not lang, language) in the meta info of the epub.

I'm going to leave this issue open as a reminder to come up with a good example of code of how to do this.
